pipeline/authz/remote_opa.go

```diff
@@ -0,0 +1,136 @@
+package authz
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/x/httpx"
+
+	"github.com/ory/oathkeeper/driver/configuration"
+	"github.com/ory/oathkeeper/helper"
+	"github.com/ory/oathkeeper/pipeline"
+	"github.com/ory/oathkeeper/pipeline/authn"
+	"github.com/ory/oathkeeper/x"
+)
+
+// AuthorizerRemoteOPAConfiguration represents a configuration for the remote_opa authorizer.
+type AuthorizerRemoteOPAConfiguration struct {
+	Remote string `json:"remote"`
+}
+
+// AuthorizerRemoteOPA implements the Authorizer interface.
+type AuthorizerRemoteOPA struct {
+	c configuration.Provider
+
+	client *http.Client
+	t      *template.Template
+}
+
+// NewAuthorizerRemoteOPA creates a new AuthorizerRemoteOPA.
+func NewAuthorizerRemoteOPA(c configuration.Provider) *AuthorizerRemoteOPA {
+	return &AuthorizerRemoteOPA{
+		c:      c,
+		client: httpx.NewResilientClientLatencyToleranceSmall(nil),
+		t:      x.NewTemplate("remote_opa"),
+	}
+}
+
+// OPA Response
+type OPAResponse struct {
+	Result struct {
+		Allow bool
+	}
+	Decision_id string
+}
+
+// OPA Policy Input
+type OPAInput struct {
+	User   string   `json:"user"`
+	Path   []string `json:"path"`
+	Method string   `json:"method"`
+}
+type OPARequest struct {
+	Input OPAInput `json:"input"`
+}
+
+// GetID implements the Authorizer interface.
+func (a *AuthorizerRemoteOPA) GetID() string {
+	return "remote_opa"
+}
+
+// Authorize implements the Authorizer interface.
+func (a *AuthorizerRemoteOPA) Authorize(_ *http.Request, session *authn.AuthenticationSession, config json.RawMessage, _ pipeline.Rule) error {
```

Comment:
Needs a test :)

Reply:
I've basically copied the remote_json test, still reading up on how Viper works and how to run the tests locally to implement some more specific tests for this authorizer.